tests/llms/test_google_palm.py

```diff
@@ -0,0 +1,86 @@
+"""Unit tests for the openai LLM class"""
+import re
+
+import pytest
+from google import generativeai
+
+from pandasai.exceptions import APIKeyNotFoundError
+from pandasai.llm.google_palm import GooglePalm
+
+
+class MockedCompletion:
+    def __init__(self, result: str):
+        self.result = result
+
+
+class TestOpenAILLM:
+    """Unit tests for the openai LLM class"""
+
+    def test_type_without_token(self):
+        with pytest.raises(APIKeyNotFoundError):
+            GooglePalm(api_key="")
+
+    def test_type_with_token(self):
+        assert GooglePalm(api_key="test").type == "google-palm"
+
+    def test_params_setting(self):
+        llm = GooglePalm(
+            api_key="test",
+            model="models/text-bison-001",
+            temperature=0.5,
+            top_p=1.0,
+            top_k=0.5,
+            max_output_tokens=64,
+        )
+
+        assert llm.model == "models/text-bison-001"
+        assert llm.temperature == 0.5
+        assert llm.top_p == 1.0
+        assert llm.top_k == 0.5
+        assert llm.max_output_tokens == 64
+
+    def test_validations(self):
+        with pytest.raises(
+            ValueError, match=re.escape("temperature must be in the range [0.0, 1.0]")
+        ):
+            GooglePalm(api_key="test", temperature=-1).call("Hello", "World")
+            GooglePalm(api_key="test", temperature=1.1).call("Hello", "World")
+
+        with pytest.raises(
+            ValueError, match=re.escape("top_p must be in the range [0.0, 1.0]")
+        ):
+            GooglePalm(api_key="test", top_p=-1).call("Hello", "World")
+            GooglePalm(api_key="test", top_p=1.1).call("Hello", "World")
+
+        with pytest.raises(
+            ValueError, match=re.escape("top_k must be in the range [0.0, 1.0]")
+        ):
+            GooglePalm(api_key="test", top_k=-1).call("Hello", "World")
+            GooglePalm(api_key="test", top_k=1.1).call("Hello", "World")
```

Comment:
Beware that these only test that **at least one** of the statements under `with pytest.raises()` raises the specified exception with the mentioned message so if your purpose was to test both the limits of `temperature`, `top_p` and `top_k` I fear here you're not doing it properly.

Reply:
Thanks for highlighting it! I've fixed it.